astropy/time/tests/test_precision.py

```diff
@@ -1,28 +1,158 @@
+import decimal
+import warnings
 import functools
+import contextlib
+from decimal import Decimal
+from datetime import datetime, timedelta
 
-import decimal
 import pytest
+from hypothesis import assume, example, given, target
+from hypothesis.extra.numpy import array_shapes, arrays
+from hypothesis.strategies import (composite, datetimes, floats, integers,
+                                   one_of, sampled_from, timedeltas, tuples)
+
 import numpy as np
-from decimal import Decimal
-from datetime import datetime
 
-import astropy.units as u
 import astropy._erfa as erfa
-from astropy.time import Time, TimeDelta
+import astropy.units as u
+from astropy.tests.helper import assert_quantity_allclose
+from astropy.time import STANDARD_TIME_SCALES, Time, TimeDelta
 from astropy.time.utils import day_frac, two_sum
 from astropy.utils import iers
-from astropy.utils.exceptions import ErfaWarning
+from astropy.utils.exceptions import ErfaError, ErfaWarning
 
-allclose_jd = functools.partial(np.allclose, rtol=2. ** -52, atol=0)
-allclose_jd2 = functools.partial(np.allclose, rtol=2. ** -52,
-                                 atol=2. ** -52)  # 20 ps atol
-allclose_sec = functools.partial(np.allclose, rtol=2. ** -52,
-                                 atol=2. ** -52 * 24 * 3600)  # 20 ps atol
+allclose_jd = functools.partial(np.allclose, rtol=np.finfo(float).eps, atol=0)
+allclose_jd2 = functools.partial(np.allclose, rtol=np.finfo(float).eps,
+                                 atol=np.finfo(float).eps)  # 20 ps atol
+allclose_sec = functools.partial(np.allclose, rtol=np.finfo(float).eps,
+                                 atol=np.finfo(float).eps * 24 * 3600)
 
-tiny = 2. ** -52
+tiny = np.finfo(float).eps
 dt_tiny = TimeDelta(tiny, format='jd')
 
 
+@pytest.fixture
+def iers_b():
+    with iers.earth_orientation_table.set(iers.IERS_B.open(iers.IERS_B_FILE)):
+        yield
+
+
+@contextlib.contextmanager
+def quiet_erfa():
+    with warnings.catch_warnings():
+        warnings.filterwarnings("ignore", category=ErfaWarning)
+        yield
+
+
+def assert_almost_equal(a, b, *, rtol=None, atol=None, label=''):
```

Comment:
This looks like it gotta move upstream to testing utils or something.

Disclaimer: I did not follow original discussions in the other PR.

Reply:
https://github.com/HypothesisWorks/hypothesis/pull/2393 summarises Hypothesis' position concisely: `target()`ing the loss of precision enables a guided search mode, and reporting the highest score makes serious problems obvious even after shrinking to a minimal example.

This specific function leverages that upstream functionality, and also integrates with Astropy's time and units functionality tightly enough that I'm not sure I could make it generic without introducing precision errors.  I'm happy to move it elsewhere in Astropy if you have a suggestion, though personally I'd be inclined to leave it as-is until an equivalent helper is needed elsewhere ("YAGNI").